### PR TITLE
Make sure references in attributes gets expanded in APIB Refract

### PR DESCRIPTION
**Pull request by pksunkara**
The attributes were getting expanded in MSON Refract in APIB AST but not in APIB Refract. This PR fixes that bug.

### CODE REVIEW

File: src/RefractAPI.cc [hunk 1]
```diff
@@ -41,15 +41,18 @@ namespace drafter {
 
     refract::IElement* DataStructureToRefract(const snowcrash::DataStructure& dataStructure)
     {
-        refract::IElement* msonElement = MSONToRefract(dataStructure);
+        refract::IElement *msonElement = NULL, *msonExpanded = NULL;
```
Comment (zdne) [inline src/RefractAPI.cc]: This line is needed. Furthermore please do declare variables on two lines without using the comma operator.

File: src/RefractAPI.cc [hunk 2]
```diff
@@ -41,15 +41,18 @@ namespace drafter {
...
 
-        if (!msonElement) {
+        msonElement = MSONToRefract(dataStructure);
```
Comment (zdne) [inline src/RefractAPI.cc]: See my comment above

**Comment by zdne** [comment]
@pksunkara 

> The attributes were getting expanded in MSON Refract in APIB AST but not in APIB Refract

Were expanded when? Please describe the issue more thoroughly, also is there a GH issue for this?  I understand this seems "more work" for you but explaining the problem is helpful even for you.

I do not like how did you moved `ExpandRefract` into `DataStructureToRefract` essentially changing the output of this function everywhere it was used. Maybe it should be called `DataStructureToExpandedRefract` then? 

What you have done is that `ElementToRefract` now always return expanded `Element::DataStructureElement` elements. Is this intended? I would much rather see this clearly separated as it was

"Side note": Also I absolutely dislike the fact that you do not use comments on the functions what so ever. This is a very bad practice not helping others to understand your code.

**Comment by zdne** [comment]
Note I am also going to discuss the lack of comments with @klokane once he is back.

**Comment by pksunkara** [comment]
> What you have done is that ElementToRefract now always return expanded Element::DataStructureElement elements. Is this intended? I would much rather see this clearly separated as it was

Apparently, this is a miscommunication and we are currently resolving this in Slack since github is down. Will update here later.

**Comment by pksunkara** [comment]
Updated.

**Comment by pksunkara** [comment]
Appveyor is failing due to github issues :disappointed:

**Comment by zdne** [comment]
For the record: We have discussed this offline – the decision is to remove expansion from the result. This will be implemented after this PR is merged.

**Pull request merged**